$OpenBSD: patch-vpx_ports_x86_cpuid_c,v 1.1 2015/02/11 00:07:47 brad Exp $

https://chromium.googlesource.com/webm/libvpx/+/9f268611472bbdfa751aedb74a306ee502e7faca

Support for extended feature flags enumeration leaf in CPUID instruction

This CL fixes an overcite with the AVX2 support CL previously
merged (Change-Id: Idc03f3fca4bf2d0afd33631ea1d3caf8fc34ec29) that
prevented runtime execution of AVX2 code in WebM.

Background:
Starting with the Sandybridge processor, the CPUID instruction was
enhanced to add various extended feature flag enumeration leaves.
Reading these leaves requires an additional input value for the CPUID
instruction which is stored in ECX. This change adds this second input
value for all ARCH_X86 and ARCH_x86_64 targets to the CPUID macros,
allowing checks of EBX bit 5 for AVX2 support. This capability will be
required moving forward to check for future processor features.

--- vpx_ports/x86_cpuid.c.orig	Thu Feb  5 20:28:42 2015
+++ vpx_ports/x86_cpuid.c	Thu Feb  5 20:28:52 2015
@@ -38,7 +38,7 @@ vpx_cpu_t vpx_x86_vendor(void) {
   int i;
 
   /* Get the Vendor String from the CPU */
-  cpuid(0, reg_eax, vs[0], vs[2], vs[1]);
+  cpuid(0, 0, reg_eax, vs[0], vs[2], vs[1]);
 
   for (i = 0; i < VPX_CPU_LAST; i++) {
     if (strncmp((const char *)vs, cpuid_vendor_list[i].vendor_string, 12) == 0)
